nsq/message.py

```diff
@@ -55,9 +56,43 @@ def __init__(self, id, body, timestamp, attempts):
         self.body = body
         self.timestamp = timestamp
         self.attempts = attempts
+        self.timeout = None
+        self.conn = None
 
         super(Message, self).__init__()
 
+    def _set_timeout(self, async_conn):
+        """
+        Sets a timeout for this message if `msg_timeout` is specified on the :class:`nsq.async.AsyncConn` instance
+            owned by the :class:`nsq.client.Client`/ :class:`nsq.reader.Reader` instance.
+
+        :param async_conn: The async_conn instance to use for the io_loop and event triggering.
+        """
+        self.conn = async_conn
+        self._clear_timeout()
+
+        def callback():
+            async_conn.trigger(event.MESSAGE_TIMEOUT, conn=self, message=self)
+            if self.timeout.is_running():
+                self._clear_timeout()
+
+        self.timeout = tornado.ioloop.PeriodicCallback(
```

Comment:
I feel like this should be implemented similar to the way the other message-based events are implemented, specifically that it triggers an event here that the parent connection is listening for, that can then call the appropriate callbacks.  I think that would reduce the state we're tracking here?